src/openrct2/actions/RideSetColourScheme.hpp

```diff
@@ -0,0 +1,73 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2018 OpenRCT2 developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#pragma once
+
+#include "../Cheats.h"
+#include "../common.h"
+#include "../core/MemoryStream.h"
+#include "../interface/Window.h"
+#include "../localisation/Localisation.h"
+#include "../localisation/StringIds.h"
+#include "../management/Finance.h"
+#include "../ride/Ride.h"
+#include "../world/Park.h"
+#include "../world/Sprite.h"
+#include "GameAction.h"
+
+struct RideSetColourSchemeAction : public GameActionBase<GAME_COMMAND_SET_COLOUR_SCHEME, GameActionResult>
+{
+private:
+    int32_t _x, _y, _z, _direction, _trackType;
+    uint16_t _newColourScheme;
+
+public:
+    RideSetColourSchemeAction()
+    {
```

Comment:
This will leave all the member fields uninitialised. Provide initialisers for them. Additionally, don't create empty constructor, make it a `= default`ed one.

Reply:
Actually, is there any need for a default ctor? It doesn't seem to be a valid use case, so perhaps it should be disabled?

Reply:
I checked and it can't be disabled due to conflicts with https://github.com/OpenRCT2/OpenRCT2/pull/8138. Please provide initialisers for member fields and make this ctor defaulted anyway.